tests/Identifier/Normalizer/IntegerDenormalizerTest.php

```diff
@@ -31,6 +31,8 @@ public function testSupportsDenormalization()
         $normalizer = new IntegerDenormalizer();
         $this->assertTrue($normalizer->supportsDenormalization('1', 'int'));
         $this->assertFalse($normalizer->supportsDenormalization([], 'int'));
+        $this->assertFalse($normalizer->supportsDenormalization('kdunglas', 'int'));
```

Comment:
This is so untrue :|. How can an identifier be a string but typed as an integer?

Reply:
Stop me if I'm wrong, but I think the identified type come from PHPDoc parsing and class reflection.

Reply:
Yes, it's linked to the property in your class. This is why it's wrong because if you typed `id` as `int` there are no reasons it should be a string ^^.

Reply:
Dirty solution, but works easier than the proper one. Maybe an issue may be open about OR conditions on identifier.

Reply:
Because the identifier should be unique, there should not be any `or` imho.